qiita_pet/handlers/study_handlers/description_handlers.py

```diff
@@ -613,6 +613,32 @@ def delete_raw_data(self, study, user, callback):
 
         callback((msg, msg_level, tab, tab_id, None))
 
+    def delete_prep_template(self, study, user, callback):
+        """Delete the selected prep template
+
+        Parameters
+        ----------
+        study : Study
+            The current study object
+        user : User
+            The current user object
+        callback : function
+            The callback function to call with the results once the processing
+            is done
+        """
+        prep_template_id = int(self.get_argument('prep_template_id'))
+
+        try:
+            PrepTemplate.delete(prep_template_id)
+            msg = ("Prep template %d has been deleted" % prep_template_id)
+            msg_level = "success"
+        except Exception as e:
+            msg = ("Couldn't remove prep template: %s" % str(e))
+            msg_level = "danger"
+
+        callback((msg, msg_level, 'raw_data_tab',
```

Comment:
This now is broken. If the removal worked successfully, the PrepTemplate with id `prep_template_id` no longer works. You just need to move the query to `PrepTemplate(prep_template_id).raw_data` before the try statement